#5 - fixed category.map is not a function error

melyasanya opened the pull request:
Fixed bug with category map by changing if check. Now we are checking if category is an array

melyasanya committed:
fixed bug

melyasanya committed:
fixed new bug

melyasanya commented:
With the fix, new bug appears, when it is impossible to choose none categories. My guess is that the functionality of choosing none category hasn't already been made, but I may be wrong and this is just another bug

nadraharr commented on src/utils/getCategoriesOption.js:
```diff
@@ -1,8 +1,10 @@
 const getCategoriesOptions = (categories) => {
-  if (categories) {
+  if (Array.isArray(categories)) {
     return categories.map((item) => (item === 'null' ? null : item))
```
Maybe it would be more logical to put this code inside of "if (categories)" block. But this version should work anyway.

melyasanya replied:
the problem is that categories can sometimes be string, and mapping string is impossible, so i need to check if categories is array to map it

vlad-khrychov commented on src/utils/getCategoriesOption.js:
Do you think it might be better to always return an array? I'm talking about the 6th line, if we get a single category, we can return an array with that category, like `[categories]`.

melyasanya replied:
Result is the same, but logically I agree with, it should better return array

melyasanya committed:
fixed bug with none categories

melyasanya commented:
In the process of fixing this bug, anothe bug appeared. It was impossible to pick none category, because when choosing none it returns null as string, not as null, so i changed it and added logic to getMatchOptions, so code will be looking for nulls in categories in database
